home-assistant: 0.116.4 -> 0.117.0 (+ additional package updates)

mweinelt opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

New upstream release. Also update and clean up dependencies.

https://github.com/home-assistant/core/releases

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

mweinelt committed:
pythonPackages.spotipy: 2.16.0 -> 2.16.1

mweinelt committed:
pythonPackages.spotipy: disable tests and cleanup

Tests require network access.

mweinelt committed:
homeassistant: 0.116.4 -> 0.117.0

mweinelt committed:
python3Packages.httpcore: 0.10.2 -> 0.12.0

Also clean up the package and enable tests.

mweinelt committed:
python3Packages.httpx: 0.14.2 -> 0.16.1

Clean up package and convert tests to pytestCheckHook.

jonringer commented:
@GrahamcOfBorg test home-assistant

jonringer commented on pkgs/development/python-modules/httpx/default.nix:
```diff
@@ -1,71 +1,61 @@
   pname = "httpx";
-  version = "0.14.2";
-  disabled = isPy27;
+  version = "0.16.1";
+  disabled = pythonOlder "3.6";
 
```
looks like a regression:

```
builder for '/nix/store/667gj5w9jcysm7pad0b4lma5hv4zkggf-python3.7-asgi-csrf-0.7.drv' failed with exit code 2; last 10 log lines:
  ImportError while importing test module '/build/source/test_asgi_csrf.py'.
  Hint: make sure your test modules/packages have valid Python names.
  Traceback:
  test_asgi_csrf.py:7: in <module>
      from httpx._content_streams import MultipartStream
  E   ModuleNotFoundError: No module named 'httpx._content_streams'
  =========================== short test summary info ============================
  ERROR test_asgi_csrf.py
  !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
  =============================== 1 error in 0.15s ===============================
cannot build derivation '/nix/store/hda4799m3wcb5mfazk2x1s77i0v8qykr-python3.7-datasette-0.46.drv': 1 dependencies couldn't be built
```

mweinelt replied:
Probably related to https://github.com/encode/httpx/commit/354c4cac1f1796ee4d8861a4c978807398852ad5.

mweinelt replied:
Reported in https://github.com/simonw/asgi-csrf/issues/18.

jonringer replied:
looks like they imported a private module... so shame on them

jonringer replied:
maybe just disable that test?

Doesn't look like it's failing on anything exported by the module

mweinelt replied:
Disabled the tests on asgi-csrf and bumped the package to the latest version for good measure.

jonringer commented:
everything else passes:

```
https://github.com/NixOS/nixpkgs/pull/101968
4 packages failed to build:
datasette python37Packages.asgi-csrf python37Packages.datasette python38Packages.asgi-csrf

11 packages built:
google-music-scripts home-assistant python27Packages.spotipy python37Packages.google-music python37Packages.httpcore python37Packages.httpx python37Packages.spotipy python38Packages.google-music python38Packages.httpcore python38Packages.httpx python38Packages.spotipy
```

mweinelt committed:
python3Packages.asgi-csrf: disable tests

The packages tests import a private module from httpx, that does not
exist anymore in newer versions.

mweinelt committed:
python3Packages.asgi-csrf: 0.7 -> 0.7.1

mweinelt commented:
Result of `nixpkgs-review pr 101968` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>datasette (python38Packages.datasette)</li>
    <li>google-music-scripts</li>
    <li>home-assistant</li>
    <li>python27Packages.spotipy</li>
    <li>python37Packages.asgi-csrf</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.spotipy</li>
    <li>python38Packages.asgi-csrf</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.spotipy</li>
  </ul>
</details>

jonringer reviewed (approved):
LGTM

Result of nixpkgs-review pr 101968 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>15 packages built:</summary>  <ul>    <li>datasette (python38Packages.datasette)</li>    <li>google-music-scripts</li>    <li>home-assistant</li>    <li>python27Packages.spotipy</li>    <li>python37Packages.asgi-csrf</li>    <li>python37Packages.datasette</li>    <li>python37Packages.google-music</li>    <li>python37Packages.httpcore</li>    <li>python37Packages.httpx</li>    <li>python37Packages.spotipy</li>    <li>python38Packages.asgi-csrf</li>    <li>python38Packages.google-music</li>    <li>python38Packages.httpcore</li>    <li>python38Packages.httpx</li>    <li>python38Packages.spotipy</li>  </ul></details>
